ecommerce/pricing/lib/pricing/services.rb

```diff
@@ -74,6 +74,7 @@ def call(cmd)
       @event_store.with_metadata({ valid_at: cmd.valid_since }) do
         @repository.with_aggregate(Product, cmd.product_id) do |product|
           product.set_price(cmd.price)
```

Comment:
The idea here is to update only the future prices calendar read model on FuturePriceSet event.
Another solution that I see for it would be, to extend the check if the event is future in the `set_price` method.
Something like: 
`    def set_price(price, valid_at)
        apply(PriceSet.new(data: { product_id: @id, price: price }))
        if valid_at
          apply(FuturePriceSet.new(data: { product_id: @id, price: price, valid_since: valid_at }))
        end
    end
`